### PR TITLE
GHC: 8.10.4 -> 8.10.5

### PR DESCRIPTION
**Pull request by TerrorJack**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
Seems like there are some references to `ghc8104` left in your PR, at least in `pkgs/development/tools/haskell/haskell-language-server/withWrapper.nix`.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11015,7 +11015,7 @@ in
 
   # Please update doc/languages-frameworks/haskell.section.md, “Our
   # current default compiler is”, if you bump this:
```
Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix:11017]: Please update the docs as well.

Comment (sternenseemann) [reply pkgs/top-level/all-packages.nix:11017]: Ah, nevermind we need to do that separately, forgot, sorry.

**Comment by maralorn** [comment]
@GrahamcOfBorg test haskell.compiler.ghc8105

**Review by maralorn — APPROVED** [review]
LGTM if CI passes

**Comment by sternenseemann** [comment]
Did successfully build `lens` with 8.10.5 on `x86_64-linux` so far.

**Comment by andir** [comment]
The commits do not conform to our contribution guidelines. Please ensure each and every commit does follow them. Specify the correct prefix etc.

**Comment by maralorn** [comment]
Huh, yeah. I mean this is about `pkgs.ghc`, so the only problem with the commit message is the capitalisation. Let’s write `ghc` instead of `GHC`.

**Comment by andir** [comment]
> Huh, yeah. I mean this is about `pkgs.ghc`, so the only problem with the commit message is the capitalisation. Let’s write `ghc` instead of `GHC`.

Also the 2nd commit (about hs lsp) should have  more then just "fixed". What was broken?

**Comment by maralorn** [comment]
That one should be squashed in my opinion.

**Comment by sternenseemann** [comment]
Note that this doesn't make us ready for `aarch64-darwin` yet — we'll need to package GHC 8.10.5 binary as well.

I first want to collect some data points on the viability of this GHC release for our package set (since stackage nightly hasn't even touched this GHC version yet). We still may have to revert this if some unforeseen regression turns up.

**Comment by maralorn** [comment]
Thank you @TerrorJack!

**Comment by domenkozar** [comment]
> Note that this doesn't make us ready for `aarch64-darwin` yet — we'll need to package GHC 8.10.5 binary as well.
> 
> I first want to collect some data points on the viability of this GHC release for our package set (since stackage nightly hasn't even touched this GHC version yet). We still may have to revert this if some unforeseen regression turns up.

Best to open an issue to track progress :)

**Pull request merged**